src/ros_filter.cpp

```diff
@@ -771,21 +771,20 @@ namespace RobotLocalization
       else
       {
         ROS_WARN_STREAM("use_control is set to true, but acceleration_limits is missing. Will use default values.");
-        accelerationLimits.resize(TWIST_SIZE, 1.0);
       }
 
-      accelerationGains.resize(TWIST_SIZE, 1.0);
       if(nhLocal_.getParam("acceleration_gains", accelerationGains))
       {
-        if(accelerationGains.size() != TWIST_SIZE)
+        const size_t size = accelerationGains.size();
+        if(size != TWIST_SIZE)
         {
           ROS_ERROR_STREAM("Acceleration gain configuration must be of size " << TWIST_SIZE <<
-            ". Provided config was of size " << accelerationGains.size() << ". All gains will be assumed to be 1.");
+            ". Provided config was of size " << size << ". All gains will be assumed to be 1.");
+          std::fill_n(accelerationGains.begin(), size, 1.0);
```

Comment:
Done. I've forced `size` to be `int`, so the instruction that fills in the vector with ones is cleaner:

`std::fill_n(accelerationGains.begin(), std::min(size, TWIST_SIZE), 1.0);`

I mean I don't need the verbose `static_cast<int>`.

This way we're actually doing the minimum number of operations:

- We only fill with ones the `size` elements of the wrong size vector given by the user, but only up to `TWIST_SIZE`
- And we resize to `TWIST_SIZE` filling with ones only for new elements; and in the case of reducing the size, the discarded elements are never filled in with ones in the previous step.

The order of the `fill_n` and `resize` instructions can be swapped, but doesn't make any difference.